runtime/tests/checker/interface_test.go

```diff
@@ -2689,3 +2689,1069 @@ func TestCheckBadStructInterface(t *testing.T) {
+		assert.Equal(t, "hello", memberConflictError.MemberName)
+		assert.Equal(t, "A", memberConflictError.ConflictingInterfaceType.QualifiedIdentifier())
+	})
+
+	t.Run("default impl from two paths", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B {
+                pub fun hello() {
+                    var a = 2
+                }
+            }
+
+            struct interface C: A, B {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		memberConflictError := &sema.InterfaceMemberConflictError{}
+		require.ErrorAs(t, errs[0], &memberConflictError)
+		assert.Equal(t, "hello", memberConflictError.MemberName)
+		assert.Equal(t, "A", memberConflictError.ConflictingInterfaceType.QualifiedIdentifier())
+	})
+
+	t.Run("overridden default impl in one path", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B: A {
+                pub fun hello() {
+                    var a = 2
+                }
+            }
+
+            struct interface C: A, B {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 2)
+
+		memberConflictError := &sema.InterfaceMemberConflictError{}
+		require.ErrorAs(t, errs[0], &memberConflictError)
+		assert.Equal(t, "hello", memberConflictError.MemberName)
+		assert.Equal(t, "A", memberConflictError.ConflictingInterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[1], &memberConflictError)
+		assert.Equal(t, "hello", memberConflictError.MemberName)
+		assert.Equal(t, "A", memberConflictError.ConflictingInterfaceType.QualifiedIdentifier())
+	})
+
+	t.Run("default impl in one path and condition in another", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B {
+                pub fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface C: A, B {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("default impl in one path and condition in another, in concrete type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B {
+                pub fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface C: A, B {}
+
+            struct D: C {}
+        `)
+
+		// The interface `C` allows to have a default implementation coming from one path,
```

Comment:
hmm, I need to look a bit more into this. It's been a while since I've added this, can't exactly remember. But it does sound off a bit. Good catch

Reply:
OK, so this is also the current behavior. A simplified version is:

```cadence
struct interface A {
    // Have default impl
    pub fun hello() {
        var a = 1
    }
}
            
struct interface B {
    // Declaration only
    pub fun hello()
}

struct C: A, B {}
```

I agree this can be further improved. But given that it's a change to the existing behavior, this is going to be a breaking change in theory if we do this. So I think it should be done in a separate PR, and preferably in stable-cadence.
i.e: no need to block supporting interface inheritance, for an improvement in default functions.

Reply:
Created an issue: https://github.com/onflow/cadence/issues/2471